### ISSUE TITLE
Limit the maximum number of requests on a downstream connection

### ISSUE THREAD

**Issue by yxue** [envoyproxy/envoy#8443]
*Description*:

Similar to the `MaxRequestsPerChild` in apache/httpd, add a configuration in the Envoy HTTP listener to limit the maximum number of requests per connection. After the number of requests hit the maximum number, the TCP connection will be gracefully closed and a new TCP connection will be initialized by the client.

**Comment by PiotrSikora** [comment]
I believe this is a duplicate of #7403.

Also related: #2941, #7852.

**Comment by dio** [comment]
@yxue closing this as it is a duplicate.

**Comment by yxue** [comment]
@PiotrSikora https://github.com/envoyproxy/envoy/issues/7403 is trying to limit the maximum streams per connection in the client side. I think this issue is for setting the limit in server side as we want the have such limit in the ingress gateway.

**Comment by thedebugger** [comment]
@yxue i was also looking for a config to limit the no of connections on the listener but didn't find any - perhaps i missed something in the docs. How did you workaround the problem?

**Comment by gokulnair** [comment]
There's `max_connection_duration` and `max_stream_duration`. Not a 100% how these 2 differ but that's the closest setting that envoy seems to have to max_requests_per_connection.

**Comment by povils** [comment]
I don;t think this is a duplicate #7403 . This issue is requesting `max_requests_per_connection` for downstream connections not upstream

**Comment by edudley** [comment]
+1 to this. 

In my situation, Chrome is establishing an http2 connection with envoy and re-using this connection until either envoy is killed or the user restarts their browser. Being able to set `max_requests_per_connection` on a listener would enable envoy to force Chrome to establish a new connection.

**Comment by sha-rath** [comment]
#14936 should resolve this and therefore can be closed.
